java/java.source.base/src/org/netbeans/modules/java/source/NoJavacHelper.java

```diff
@@ -69,35 +68,20 @@ public void run() {
                 w.visit(Opcodes.V1_8, Opcodes.ACC_ABSTRACT | Opcodes.ACC_PUBLIC, "com/sun/tools/javac/code/Scope$WriteableScope", null, "com/sun/tools/javac/code/Scope", null);
                 byte[] classData = w.toByteArray();
 
-                String JavaVersion = System.getProperty("java.specification.version"); //NOI18N
-                boolean isJdkVer8OrBelow = true;
-                if (!JavaVersion.startsWith("1.")) {   //NOI18N
-                    isJdkVer8OrBelow = false;
-                }
-                if (isJdkVer8OrBelow) {
-                    try {
-                        Field theUnsafe = Unsafe.class.getDeclaredField("theUnsafe"); //NOI18N
-                        theUnsafe.setAccessible(true);
-                        Unsafe unsafe = (Unsafe) theUnsafe.get(null);
+                String[] javaVersionElements = System.getProperty("java.version").split("\\.");
+                int major = Integer.parseInt(javaVersionElements[1]);
```

Comment:
This looks only sane on the old version scheme 1.X.Y. As shown above, the version scheme is shifting...